core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ShaclSail.java

```diff
@@ -307,11 +322,22 @@ public void shutDown() throws SailException {
 			shapesRepo = null;
 		}
 
+		if (executorService != null) {
+			executorService.shutdown();
```

Comment:
We shutdown the executorService in shutDown(). This makes it very important to remember to shut down the sail.

Reply:
Actually wonder if we shouldn't make the sails closable to indicate that they need to be closed/shutdown?